var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -136,4 +135,68 @@ def cmake_args(self):
             args.append("-DBUILD_STATS=ON")
         else:
             args.append("-DBUILD_STATS=OFF")
+
+        if self.run_tests:
+            args.append("-DBUILD_TESTS=ON")
+            args.append("-DINTEGRATION_TESTS=ON")
+        else:
+            args.append("-DBUILD_TESTS=OFF")
+            args.append("-DINTEGRATION_TESTS=OFF")
```

Comment:
This can be simplified to the following since your package inherits from `CMakePackage`:

```suggestion
        args.append(self.define("BUILD_TESTS", self.run_tests))
        args.append(self.define("INTEGRATION_TESTS", self.run_tests))
```

This is discussed some in:
- https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake
- https://spack.readthedocs.io/en/latest/spack.build_systems.html#module-spack.build_systems.cmake  (especially for `self.define` but also check out `self.define_from_variant` that could be used to simplify lines 93-137)